libs/cardano-ledger-test/src/Test/Cardano/Ledger/Constrained/TypeRep.hs

```diff
@@ -294,128 +297,103 @@ stringR = ListR CharR
+  WitVKeyR{}             -> IsTypeable
+  TxAuxDataR{}           -> IsTypeable
+  LanguageR{}            -> IsTypeable
+  LedgerStateR{}         -> IsTypeable
+  TxR{}                  -> IsTypeable
+  ScriptIntegrityHashR{} -> IsTypeable
+  AuxiliaryDataHashR{}   -> IsTypeable
+  BootstrapWitnessR{}    -> IsTypeable
+  SigningKeyR{}          -> IsTypeable
+  TxWitsR{}              -> IsTypeable
+  PayHashR{}             -> IsTypeable
+  IntegerR{}             -> IsTypeable
+  ScriptsNeededR{}       -> IsTypeable
+  ScriptPurposeR{}       -> IsTypeable
+  TxBodyR{}              -> IsTypeable
+  ShelleyTxCertR{}       -> IsTypeable
+  ConwayTxCertR{}        -> IsTypeable
+  MIRPotR{}              -> IsTypeable
+  IsValidR{}             -> IsTypeable
+  ExUnitsR{}             -> IsTypeable
+  TagR{}                 -> IsTypeable
+  DataHashR{}            -> IsTypeable
+  PCredR{}               -> IsTypeable
+  NetworkR{}             -> IsTypeable
+  RdmrPtrR{}             -> IsTypeable
+  DataR{}                -> IsTypeable
+  DatumR{}               -> IsTypeable
+  KeyPairR{}             -> IsTypeable
+  ScriptR{}              -> IsTypeable
+  ScriptHashR{}          -> IsTypeable
+  TxCertR{}              -> IsTypeable
+  RewardAcntR{}          -> IsTypeable
+  ValidityIntervalR{}    -> IsTypeable
+  AssetNameR {}          -> IsTypeable
+  WitnessesFieldR{}      -> IsTypeable
+  MultiAssetR{}          -> IsTypeable
+  PolicyIDR{}            -> IsTypeable
+  CharR{}                -> IsTypeable
+  RationalR{}            -> IsTypeable
+  CoinR{}                -> IsTypeable
+  EpochR{}               -> IsTypeable
+  AddrR{}                -> IsTypeable
+  CredR{}                -> IsTypeable
+  VCredR{}               -> IsTypeable
+  PoolHashR{}            -> IsTypeable
+  WitHashR{}             -> IsTypeable
+  GenHashR{}             -> IsTypeable
+  GenDelegHashR{}        -> IsTypeable
+  VHashR{}               -> IsTypeable
+  PoolParamsR{}          -> IsTypeable
+  NewEpochStateR{}       -> IsTypeable
+  IntR{}                 -> IsTypeable
+  FloatR{}               -> IsTypeable
+  NaturalR{}             -> IsTypeable
+  Word64R{}              -> IsTypeable
+  TxInR{}                -> IsTypeable
+  UnitR{}                -> IsTypeable
+  ProtVerR{}             -> IsTypeable
+  ValueR{}               -> IsTypeable
+  UTxOR{}                -> IsTypeable
+  TxOutR{}               -> IsTypeable
+  PParamsR{}             -> IsTypeable
+  PParamsUpdateR{}       -> IsTypeable
+  DeltaCoinR{}           -> IsTypeable
+  GenDelegPairR{}        -> IsTypeable
+  FutureGenDelegR{}      -> IsTypeable
+  PPUPStateR{}           -> IsTypeable
+  PtrR{}                 -> IsTypeable
+  IPoolStakeR{}          -> IsTypeable
+  SnapShotsR{}           -> IsTypeable
+  RewardR{}              -> IsTypeable
+  SlotNoR{}              -> IsTypeable
+  SizeR{}                -> IsTypeable
+  (repTypeable -> IsTypeable) :-> (repTypeable -> IsTypeable) -> IsTypeable
```

Comment:
I don't think I've seen this kind of pattern matching before, what does it do?

Reply:
This matches the constructor and ignores any arguments that would be bound in the pattern. So `Just{}` is equivalent to `Just _`.

Reply:
Ah, I mean this part of the pattern match `(repTypeable -> IsTypeable)`

Reply:
That's a view pattern, it means roughly `Just (negate -> a) = e ~ Just _x = let a = negate _x in e`